devices/common/diy.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+#=================================================
+shopt -s extglob
+
+sed -i '$a src-git kiddin9 https://github.com/kiddin9/kwrt-packages.git;master' feeds.conf.default
+sed -i "/telephony/d" feeds.conf.default
+
+sed -i "s?targets/%S/packages?targets/%S/\$(LINUX_VERSION)?" include/feeds.mk
+
+sed -i '/	refresh_config();/d' scripts/feeds
+
+./scripts/feeds update -a
+./scripts/feeds install -a -p kiddin9 -f
+./scripts/feeds install -a
+
+rm -rf package/base-files
+mv -f feeds/kiddin9/base-files package/
+
+echo "$(date +"%s")" >version.date
+sed -i '/$(curdir)\/compile:/c\$(curdir)/compile: package/opkg/host/compile' package/Makefile
+sed -i "s/DEFAULT_PACKAGES:=/DEFAULT_PACKAGES:=luci-app-advancedplus luci-app-firewall luci-app-opkg luci-app-upnp \
+luci-app-wizard luci-base luci-compat luci-lib-ipkg luci-lib-fs \
+coremark wget-ssl curl autocore htop nano zram-swap kmod-lib-zstd kmod-tcp-bbr bash openssh-sftp-server block-mount resolveip ds-lite swconfig luci-app-fan luci-app-fileassistant /" include/target.mk
+
+sed -i "s/procd-ujail//" include/target.mk
+sed -i "s/procd-seccomp//" include/target.mk
+
+sed -i "s/^.*vermagic$/\techo '1' > \$(LINUX_DIR)\/.vermagic/" include/kernel-defaults.mk
+
+status=$(curl -H "Authorization: token $REPO_TOKEN" -s "https://api.github.com/repos/kiddin9/kwrt-packages/actions/runs" | jq -r '.workflow_runs[0].status')
+echo "$status"
+while [[ "$status" == "in_progress" || "$status" == "queued" ]];do
```

Comment:
According to the documentation (https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28) the status can also be set to several other values, including `pending` and `requested`. 
I think the while loop body should continue executing when the status is equal to `pending` and `requested` apart from `in_progress` and `queued`.